src/main/java/com/leverx/springapp/demo/controller/UserController.java

```diff
@@ -0,0 +1,52 @@
+package com.leverx.springapp.demo.controller;
+
+import com.leverx.springapp.demo.model.User;
+import com.leverx.springapp.demo.repository.UserRepository;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    private UserRepository userRepository;
+
+    public UserController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @PostMapping
+    public ResponseEntity createUser(@Valid @RequestBody User user) {
+        userRepository.save(user);
+        URI location = ServletUriComponentsBuilder
+                .fromCurrentRequest().path("/{id}")
+                .buildAndExpand(user.getId()).toUri();
+        return ResponseEntity.created(location).build();
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity get(@PathVariable("id") Long id) {
+        if (userRepository.findById(id).isPresent()) {
```

Comment:
@blackavgn findById and then, if present findById? Please fix it. Also, optional already have 'pretty' alternative for 'if (x.isPresent) then' constructions. Please check 'map' and 'orElse' operations.